taf/developer_tool.py

```diff
@@ -155,76 +276,52 @@ def build_auth_repo(
-                for existing_role_name, role_keys in yubikeys.items():
-                    if existing_role_name == role_name:
-                        continue
-                    print(f"Existing keys for role {existing_role_name} are:\n")
-                    for key_and_cert in role_keys.values():
-                        key, cert_cn = key_and_cert
-                        key_id_certs[key["keyid"]] = cert_cn
-                        print(f"{cert_cn} id: {key['keyid']}")
-                existing_keyid = input("\nEnter existing Yubikey's id and press ENTER ")
-                try:
-                    existing_key = get_key(existing_keyid)
-                    cert_cn = key_id_certs[existing_keyid]
-                except UnknownKeyError:
-                    pass
-    if not use_existing:
-        input("Please insert a new YubiKey and press ENTER.")
-        serial_num = yk.get_serial_num()
-        while serial_num in yubikeys[role_name]:
-            print(f"Yubikey with serial number {serial_num} is already in use.\n")
-            input("Please insert new YubiKey and press ENTER.")
-            serial_num = yk.get_serial_num()
-
-        pin = get_pin_for(key_name)
-
-        cert_cn = input("Enter key holder's name: ")
-
-        print("Generating keys, please wait...")
-        pub_key_pem = yk.setup(pin, cert_cn, cert_exp_days=EXPIRATION_INTERVAL).decode(
-            "utf-8"
+    from taf.repository_tool import yubikey_signature_provider
+
+    registered = False
+    while not registered:
+        print(f"Registering keys for {key_name}")
+        use_existing = click.confirm("Do you want to reuse already set up Yubikey?")
+
+        if not use_existing:
+            if not click.confirm(
+                "WARNING - this will delete everything from the inserted key. Proceed?"
+            ):
+                continue
+
+        key, serial_num = yk.yubikey_prompt(
+            key_name,
+            role_name,
+            taf_repo=None,
+            registering_new_key=True,
+            creating_new_key=not use_existing,
+            loaded_yubikeys=yubikeys,
+            pin_confirm=True,
+            pin_repeat=True,
         )
 
-        key = import_rsakey_from_pem(pub_key_pem, scheme)
-
-        cert_path = Path(certs_dir, f"{key['keyid']}.cert")
-        with open(cert_path, "wb") as f:
-            f.write(yk.export_piv_x509())
+        if not use_existing:
+            _setup_new_yubikey(serial_num, certs_dir, scheme)
 
-    # set Yubikey expiration date
-    role_obj.add_verification_key(key, expires=YUBIKEY_EXPIRATION_DATE)
-    role_obj.add_external_signature_provider(
-        key, partial(signature_provider, key["keyid"], cert_cn)
-    )
-    yubikeys[role_name][serial_num] = (key, cert_cn)
+        registered = True
+        # set Yubikey expiration date
+        # add_yubikey_external_signature_provider
+        role_obj.add_verification_key(key, expires=YUBIKEY_EXPIRATION_DATE)
```

Comment:
`key` is `None` if I choose to setup a new yubikey

Reply:
There was a return missing in `setup_new_yubikey`.